content/en/docs/tasks/administer-cluster/coredns.md

```diff
@@ -34,51 +37,33 @@ For manual deployment or replacement of kube-dns, see the documentation at the
 
 ### Upgrading an existing cluster with kubeadm
 
-In Kubernetes version 1.10 and later, you can also move to CoreDNS when you use `kubeadm` to upgrade
-a cluster that is using `kube-dns`. In this case, `kubeadm` will generate the CoreDNS configuration
+In Kubernetes version 1.21, support for `kube-dns` is removed from kubeadm.
+You can move to CoreDNS when you use `kubeadm` to upgrade a cluster that is
+using `kube-dns`. In this case, `kubeadm` will generate the CoreDNS configuration
 ("Corefile") based upon the `kube-dns` ConfigMap, preserving configurations for
 stub domains, and upstream name server.
 
-If you are moving from kube-dns to CoreDNS, make sure to set the `CoreDNS` feature gate to `true`
-during an upgrade. For example, here is what a `v1.11.0` upgrade would look like:
-```
-kubeadm upgrade apply v1.11.0 --feature-gates=CoreDNS=true
-```
-
-In Kubernetes version 1.13 and later the `CoreDNS` feature gate is removed and CoreDNS
-is used by default.
-
-In versions prior to 1.11 the Corefile will be **overwritten** by the one created during upgrade.
-**You should save your existing ConfigMap if you have customized it.** You may re-apply your
-customizations after the new ConfigMap is up and running.
-
-If you are running CoreDNS in Kubernetes version 1.11 and later, during upgrade,
-your existing Corefile will be retained.
-
-In Kubernetes version 1.21, support for `kube-dns` is removed from kubeadm.
-
 ## Upgrading CoreDNS
 
-CoreDNS is available in Kubernetes since v1.9.
-You can check the version of CoreDNS shipped with Kubernetes and the changes made to CoreDNS [here](https://github.com/coredns/deployment/blob/master/kubernetes/CoreDNS-k8s_version.md).
+You can check the version of CoreDNS shipped with Kubernetes and the changes
+made to CoreDNS [here](https://github.com/coredns/deployment/blob/master/kubernetes/CoreDNS-k8s_version.md).
 
-CoreDNS can be upgraded manually in case you want to only upgrade CoreDNS or use your own custom image.
-There is a helpful [guideline and walkthrough](https://github.com/coredns/deployment/blob/master/kubernetes/Upgrading_CoreDNS.md) available to ensure a smooth upgrade.
+During cluster upgrade, your existing Corefile will be retained.
```

Comment:
Is this true and appropriate:
```suggestion
If your Kubernetes cluster already uses CoreDNS, retain the existing `Corefile` (the
configuration file for CoreDNS) when you upgrade. If you use the `kubeadm` tool then
the upgrade takes care of retaining the existing configuration automatically.
```
?

Reply:
Agreed. The verbose version is more accurate as it reads.